### ISSUE TITLE
Cello never stop

### ISSUE THREAD

**Issue by VitoVan** [alda-lang/alda#120]
~ $ alda repl
```
cello: b-8
```
Then the cello will never stop.

It seems to be the number definition in [alda.bnf](https://github.com/alda-lang/alda/blob/master/grammar/alda.bnf#L37):
```
number                  = #"-?[0-9]+"
```
This makes this:
```clojure
(alda.parser/parse-input "cello: b-8")
```
to
```clojure
(alda.lisp/score 
 (alda.lisp/part {:names ["cello"]} 
                 (alda.lisp/note
                  (alda.lisp/pitch :b)
                  (alda.lisp/duration (alda.lisp/note-length -8)))))
```
Which is supposed to be b-flat with note-length=8, instead of b with note-length=-8.

Then I change number definition in alda.bnf to
```
number                  = #"[0-9]+"
```
then this
```clojure
(alda.parser/parse-input "cello: b-8")
```
will generate
```clojure
(alda.lisp/score
 (alda.lisp/part {:names ["cello"]}
                 (alda.lisp/note
                  (alda.lisp/pitch :b :flat)
                  (alda.lisp/duration (alda.lisp/note-length 8)))))
```
It seems fine, and works like a charm with ```boot alda -x "repl"```.
If we don't need any negative number in alda, it will be more convenient to just make number as ```#"[0-9]+"```.

**Comment by crisptrutski** [comment]
Yowser! Thanks for catching this. Looking over the grammar, I'm uncertain why negative numbers are supported. Potentially negative octaves make sense. (ping @daveyarwood)

If that's the case, I'd propose introducing a new `negative-number` node to use there.

@VitoVan feel free to open a PR here, otherwise I'll patch this tonight.

**Comment by crisptrutski** [comment]
Perhaps adding a `pos?` assert around note longs would be good hygeine too. Even then, not stopping reminds me of #27 - it feels like we're missing something to make MIDI less error prone in general.

**Comment by VitoVan** [comment]
@crisptrutski It seems the work is more than only changing the definition of ```number```, so maybe it's better to leave it to you.

And, thank you for asking! I'll open a PR next time (if I have the ability to solve it of course).

Wondering when is your "tonight", the sun has gone sleep on my planet :smile_cat: .

**Comment by crisptrutski** [comment]
I'm GMT +2, South Africa. Will be home in about 8 hours, but @daveyarwood might beat me to the punch

**Comment by daveyarwood** [comment]
@VitoVan Thanks for reporting this! I had no idea it was happening. :astonished: 

I think `cello: b-8` used to work before I made a change to the grammar ([in this recent commit](https://github.com/alda-lang/alda/commit/341e55d21bb4c2773b6460bf17df54ab3b7542c6)) explicitly making "flat" and "sharp" their own tokens, rather than being part of the regex for a note. Regexes are greedy, but rules are not, so now apparently minuses are gobbled up by the `note-length` rule.

Note lengths definitely should not be negative, so an easy fix would be to add a "positive-number" rule, and point note-length at that instead of the "number" rule, which can include negative numbers.

The reason that negative numbers are a usable thing in the grammar actually has to do with octaves -- in scientific pitch notation, it's actually possible to have notes in negative octaves, and in fact, the first 12(?) MIDI notes are in octave -1 for some weird reason. You can play around with this in an Alda REPL by typing `piano: o4 c o3 c o2 c o1 c o0 c o-1 c`

**Comment by daveyarwood** [comment]
```
> piano: o-1 c
p> :map
{:events
 #{{:offset 0,
    :instrument "piano-zWIrN",
    :volume 1.0,
    :track-volume 0.7874015748031497,
    :panning 0.5,
    :midi-note 0,
    :pitch 8.175798915643707,
    :duration 450.0}},
 :markers {:start 0},
 :instruments
 {"piano-zWIrN"
  {:octave -1,
   :current-offset {:offset 500.0},
   :key-signature {},
   :config {:type :midi, :patch 1},
   :duration 1,
   :volume 1.0,
   :last-offset {:offset 0},
   :id "piano-zWIrN",
   :quantization 0.9,
   :tempo 120,
   :panning 0.5,
   :current-marker :start,
   :stock "midi-acoustic-grand-piano",
   :track-volume 0.7874015748031497}}}
p>
```

MIDI note 0 = C-1

**Comment by daveyarwood** [comment]
Fixed in alda 0.10.4, which I pushed out just now.
